subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2498,8 +2500,15 @@ static void le_advertising_report(struct pdu_data *pdu_data, u8_t *b,
 #endif /* CONFIG_BT_CTLR_EXT_SCAN_FP */
 	s8_t *prssi;
 
-	extra = &b[offsetof(struct node_rx_pdu, pdu) +
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+	struct node_rx_pdu *node_rx;
+
+	node_rx = (struct node_rx_pdu *)b;
+	extra = (u8_t *) &(node_rx->hdr.rx_ftr.param);
```

Comment:
Several points:
1. `rx_ftr` no longer makes sense as a name.
2. rssi and the following field is an implicit data structure, just happening to use the `rx_ftr`'s memory. Thus rssi & co, should become an explicit data structure and unionized with `rx_ftr`.
3. `extra` as a name makes little sense. It should be replaced as part of step 2.

I think we should address these comments as a follow-up PR.

Reply:
Agreed this should be done in a subsequent PR.

Reply:
agreed on all the above.

Reply:
no space in between ```(u8_t *) &(```